lib/linux_admin/hosts.rb

```diff
@@ -36,6 +38,20 @@ def update_entry(address, hostname, comment = nil)
       end
     end
 
+    def hostname=(name)
+      if cmd?("hostnamectl")
+        run!(cmd('hostnamectl'), :params => ['set-hostname', name])
+      else
+        File.write("/etc/hostname", name)
+        run!(cmd('hostname'), :params => {:file => "/etc/hostname"})
+      end
+    end
+
+    def hostname
+      result = run(cmd("hostname"))
```

Comment:
I'm not sure about what will be on output.  It depends really on what `Open3.capture3` returns here (https://github.com/ManageIQ/awesome_spawn/blob/master/lib/awesome_spawn.rb#L119) and what `hostname` returns in whatever failure case happened.